data-prepper-plugins/kinesis-source/src/integrationTest/java/org/opensearch/dataprepper/plugins/kinesis/source/KinesisSourceIT.java

```diff
@@ -0,0 +1,286 @@
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.ACKNOWLEDGEMENT_SET_FAILURES_METRIC_NAME;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.ACKNOWLEDGEMENT_SET_SUCCESS_METRIC_NAME;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.KINESIS_RECORD_PROCESSED;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.KINESIS_RECORD_PROCESSING_ERRORS;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.KINESIS_CHECKPOINT_FAILURES;
+import static org.opensearch.dataprepper.plugins.kinesis.source.processor.KinesisRecordProcessor.KINESIS_STREAM_TAG_KEY;
+
+public class KinesisSourceIT {
+    private static final String AWS_REGION = "tests.kinesis.source.aws.region";
+    private static final String STREAM_NAME_PREFIX = "Stream1";
+    private static final String PIPELINE_NAME = "pipeline-id";
+    private static final Duration CHECKPOINT_INTERVAL = Duration.ofMillis(0);
+    private static final String codec_plugin_name = "ndjson";
+    private static final String LEASE_TABLE_PREFIX = "kinesis-lease-table";
+    private static final int NUMBER_OF_RECORDS_TO_ACCUMULATE = 10;
+
+    @Mock
+    private AcknowledgementSetManager acknowledgementSetManager;
+
+    @Mock
+    private KinesisClientFactory kinesisClientFactory;
+
+    @Mock
+    private Buffer<Record<Event>> buffer;
+
+    @Mock
+    private KinesisSourceConfig kinesisSourceConfig;
+
+    @Mock
+    private AwsAuthenticationConfig awsAuthenticationConfig;
+
+    @Mock
+    private KinesisStreamConfig kinesisStreamConfig;
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private PluginFactory pluginFactory;
+
+    @Mock
+    private PipelineDescription pipelineDescription;
+
+    @Mock
+    private KinesisLeaseConfigSupplier kinesisLeaseConfigSupplier;
+
+    @Mock
+    private KinesisLeaseConfig kinesisLeaseConfig;
+
+    @Mock
+    private WorkerIdentifierGenerator workerIdentifierGenerator;
+
+    @Mock
+    private KinesisLeaseCoordinationTableConfig kinesisLeaseCoordinationTableConfig;
+
+    @Mock
+    private InputCodec codec;
+
+    @Mock
+    private Counter acknowledgementSetSuccesses;
+
+    @Mock
+    private Counter acknowledgementSetFailures;
+
+    @Mock
+    private Counter recordsProcessed;
+
+    @Mock
+    private Counter recordProcessingErrors;
+
+    @Mock
+    private Counter checkpointFailures;
+
+    private KinesisClient kinesisClient;
+
+    private DynamoDbClient dynamoDbClient;
+
+    private KinesisIngester kinesisIngester;
+
+    @BeforeEach
+    void setup() {
+        this.acknowledgementSetManager = mock(AcknowledgementSetManager.class);
```

Comment:
You might consider `MockitoAnnotations.openMocks(this)` for initializing mocked instances

Reply:
Thanks @chenqi0805! I will address this.